### PR TITLE
chore(generation): Update The User-Agent Header

### PR DESCRIPTION
**Pull request by osama-salman99**
* [x] Code is up-to-date with the `main` branch.
* [x] You've successfully built and run the tests locally.
* [ ] There are new or updated unit tests validating the change.

### :pencil: Description
- Update the user-agent header

### :link: Related Issues
N/A

### CODE REVIEW

File: core/src/client/Client.ts [hunk 1]
```diff
@@ -47,5 +47,10 @@ interface ClientConfigurations {
   requestTimeout?: number
 }
 
+interface ExtendedClientConfiguration extends ClientConfigurations {
```
Comment (osama-salman99) [inline core/src/client/Client.ts:50]: I was hesitant about what to name this interface, so I left this as a placeholder

File: generator/src/main/resources/templates/expediagroup-sdk/api/api.mustache [hunk 2]
```diff
@@ -18,10 +18,10 @@ export class {{classname}} extends Client {
     private static readonly nodeVersion: string = process.version
     private static readonly operatingSystemName: string = platform()
     private static readonly operatingSystemVersion: string = release()
-    private static readonly userAgent: string = `expediagroup-sdk-node-{{namespace}}/{{artifactVersion}} (Node ${this.nodeVersion} ${this.operatingSystemName} ${this.operatingSystemVersion})`
+    private static readonly userAgent: string = `${this.sdkTitle}/{{artifactVersion}} (NodeJS ${this.nodeVersion} ${this.operatingSystemName} ${this.operatingSystemVersion})`
```
Comment (anssari1) [inline generator/src/main/resources/templates/expediagroup-sdk/api/api.mustache:21]: Node.js

Comment (osama-salman99) [reply generator/src/main/resources/templates/expediagroup-sdk/api/api.mustache]: Done ✅

File: core/src/client/Client.ts (hunk 1 above)
Comment (anssari1) [reply core/src/client/Client.ts:50]: ClientConfiguration & { userAgent: string }

**Review by anssari1 — APPROVED** [review]

**Pull request merged**